src/Game.ts

```diff
@@ -19,41 +24,69 @@ class Game {
         this.blackClock.render();
         this.whiteClock = new Clock('white', 15, 0, 'whiteClock');
         this.whiteClock.render();
+        this.promo = new Promo();
     }
-    
-    afterMove(field: Field, move: string) {
 
+    afterMove(field: Field, move: string) {
         const newField = this.board.getField(parseInt(move[0]), parseInt(move[2]));
         this.movePiece(field, newField);
-        this.promotePawn(newField);
         this.updateEnpassantStatus();
         this.movePiece(field, move);
         this.isCheck();
-        this.changeTurn();
-        this.changeClock();
-        // Logika która powinna znajdować sie po ruchu znajduje się tutaj,
-        // oczywiście chodzi tutaj o wywołania odpowiednich funkcji tylko :)
-        // czyli np. sprawdzenie czy jest szach, mat, pat, zmiana tury itp.
+        if (newField.piece?.display?.match(/pawn/i) && ['0','7'].includes(move[0])) {
+            this.promotePawn(newField)
+        } else {
+            this.changeTurn();
+            this.changeClock();
+        }
     }
 
-
     promotePawn(newField: Field): void {
-        const color  = this.turn === 'white' ? 0 : 7
-        // const field = newField.piece?.side === 'white' ? 0 : 7 
-        for (let y = 0; y < this.board.boardSize; y++) {
-            if (this.board.fields[color][y].piece instanceof Pawn) {
-                this.board.fields[color][y].piece = new Queen(this.turn);
-                this.board.fields[color][y].piece?.render(newField);
+        if(newField.piece instanceof Pawn) {
+            const color = this.turn === 'white' ? 0 : 7;
+            for (let y = 0; y < this.board.boardSize; y++) {
+                if (this.board.fields[color][y].piece?.display?.match(/pawn/i) && ['0','7'].includes(color.toString())) {
+                    this.promo.promoScreenTrigger();
+                    const possiblePromoPieces = document.querySelectorAll('.promo div');
+                    possiblePromoPieces.forEach(piece => {
+                        piece.addEventListener('click', () => {
```

Comment:
Cały problem wystepuje w tym miejscu, dodajemy do danego pola eventListenera i w tym momencie on zawsze jest aktywny wiec przt kolejnej promocji on sie znowu wywołuje i w miedzy czasie mamy zmiane tury co powoduje to że dzieją się niestety takie rzeczy, trzeba usunac te eventListenery jak juz nie beda potrzebne